expression/expressions/exists_subquery.go

```diff
@@ -50,26 +50,18 @@ func (es *ExistsSubQuery) String() string {
 
 // Eval implements the Expression Eval interface.
 func (es *ExistsSubQuery) Eval(ctx context.Context, args map[interface{}]interface{}) (interface{}, error) {
-	if es.Sel.Value != nil {
-		return true, nil
+	if !es.Sel.UseOuterQuery && es.Sel.Value != nil {
+		return es.Sel.Value.(int) > 0, nil
 	}
 
-	p, err := es.Sel.Plan(ctx)
+	rows, err := es.Sel.EvalRows(ctx, args, 1)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	defer p.Close()
 
-	r, err := p.Next(ctx)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	if r != nil {
-		es.Sel.Value = r
-		return true, nil
-	}
-
-	return false, nil
+	// can only save length here
```

Comment:
Why not `es.Sel.Value = len(rows) > 0` and at line 54 `return es.Sel.Value.(bool), nil`?